NullPointerException in VolumeRemoveMojo

### Description
If I try to run the goal docker:volume-remove, and the plugin could not find any volume names. It throws a NullPointerException.
In the debug log you could see that the parameter volumes are not set, if no volume configuration exists in the module. So the mojo throws a NullPointerException.
Caused by: java.lang.NullPointerException
	at io.fabric8.maven.docker.VolumeRemoveMojo.executeInternal(VolumeRemoveMojo.java:26)
	at io.fabric8.maven.docker.AbstractDockerMojo.execute(AbstractDockerMojo.java:223)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
If the maven module contains a voulme configuration. It removes the volume.
I think you shuld implement a NPE check to avoid this issue.

If configuration exists in project:
...
[DEBUG]   (f) session = org.apache.maven.execution.MavenSession@1f12e153
[DEBUG]   (f) settings = org.apache.maven.execution.SettingsAdapter@5a101b1c
[DEBUG]   (f) skip = false
[DEBUG]   (f) skipExtendedAuth = false
[DEBUG]   (f) skipMachine = false
[DEBUG]   (f) useColor = true
[DEBUG]   (f) verbose = true
[DEBUG]   (f) name = oradata
[DEBUG]   (f) opts = {o=uid=1000}
[DEBUG]   (f) volumes = [io.fabric8.maven.docker.config.VolumeConfiguration@5e2a3040]
[DEBUG] -- end configuration --

If no configuration exists in project:
...
[DEBUG]   (f) removeVolumes = false
[DEBUG]   (f) session = org.apache.maven.execution.MavenSession@47289387
[DEBUG]   (f) settings = org.apache.maven.execution.SettingsAdapter@114a85c2
[DEBUG]   (f) skip = false
[DEBUG]   (f) skipExtendedAuth = false
[DEBUG]   (f) skipMachine = false
[DEBUG]   (f) useColor = true
[DEBUG]   (f) verbose = true
[DEBUG] -- end configuration --

### Info

* d-m-p version : 0.24.0 and 0.26.0
* Maven version (`mvn -v`) :
```
Apache Maven 3.3.9 (bb52d8502b132ec0a5a3f4c09453c07478323dc5; 2015-11-10T17:41:47+01:00)
Maven home: /opt/idea/plugins/maven/lib/maven3
Java version: 1.8.0_171, vendor: Oracle Corporation
Java home: /opt/oracle_java/jdk1.8.0_171/jre
```
* Docker version :
Client:
 Version:           18.06.1-ce
 API version:       1.38
 Go version:        go1.10.3
 Git commit:        e68fc7a
 Built:             Tue Aug 21 17:24:51 2018
 OS/Arch:           linux/amd64
 Experimental:      false

Server:
 Engine:
  Version:          18.06.1-ce
  API version:      1.38 (minimum version 1.12)
  Go version:       go1.10.3
  Git commit:       e68fc7a
  Built:            Tue Aug 21 17:23:15 2018
  OS/Arch:          linux/amd64
  Experimental:     false

* If it's a bug, how to reproduce :
1. Create a simple maven project with two modules
2. Add docker plugin configuration with volumes in the first module
4. Execute the docker:volume-remove command in the parent pom

Result: The plugin will fail on execute the goal in the parent pom or in the second module.

Possible fix to VolumeRemoveMojo.java could be:
if(getVolumes() != null){
for ( VolumeConfiguration volume : getVolumes()) {
         log.info("Removing volume %s", volume.getName());
         volService.removeVolume(volume.getName());
      }
}else{
log.info("No volume configuration found");
}

--- a/src/main/java/io/fabric8/maven/docker/VolumeCreateMojo.java
+++ b/src/main/java/io/fabric8/maven/docker/VolumeCreateMojo.java
@@ -10,22 +10,26 @@ import org.apache.maven.plugins.annotations.LifecyclePhase;
 import org.apache.maven.plugins.annotations.Mojo;
 
 /**
- *  Mojo to create named volumes, useful for preparing integration tests
+ * Mojo to create named volumes, useful for preparing integration tests
  *
- *  @author Tom Burton
- *  @version Dec 15, 2016
+ * @author Tom Burton
+ * @version Dec 15, 2016
  */
 @Mojo(name = "volume-create", defaultPhase = LifecyclePhase.PRE_INTEGRATION_TEST)
 public class VolumeCreateMojo extends AbstractDockerMojo {
 
-   @Override
-   protected void executeInternal(ServiceHub serviceHub) throws DockerAccessException, MojoExecutionException {
-      VolumeService volService = serviceHub.getVolumeService();
+    @Override
+    protected void executeInternal(ServiceHub serviceHub) throws DockerAccessException, MojoExecutionException {
+        VolumeService volService = serviceHub.getVolumeService();
 
-      for (VolumeConfiguration volume : getVolumes()) {
-         log.info("Creating volume '%s'", volume.getName());
-         volService.createVolume(volume);
-      }
-   }
+        if (getVolumes() != null) {
+            for (VolumeConfiguration volume : getVolumes()) {
+                log.info("Creating volume '%s'", volume.getName());
+                volService.createVolume(volume);
+            }
+        } else {
+            log.info("No volume configuration found.");
+        }
+    }
 
 }

--- a/src/main/java/io/fabric8/maven/docker/VolumeCreateMojo.java
+++ b/src/main/java/io/fabric8/maven/docker/VolumeCreateMojo.java
@@ -20,15 +20,15 @@ public class VolumeCreateMojo extends AbstractDockerMojo {
 
     @Override
     protected void executeInternal(ServiceHub serviceHub) throws DockerAccessException, MojoExecutionException {
+        if (getVolumes() == null) {
+            log.info("No volume configuration found.");
+            return;
+        }
         VolumeService volService = serviceHub.getVolumeService();
 
-        if (getVolumes() != null) {
-            for (VolumeConfiguration volume : getVolumes()) {
-                log.info("Creating volume '%s'", volume.getName());
-                volService.createVolume(volume);
-            }
-        } else {
-            log.info("No volume configuration found.");
+        for (VolumeConfiguration volume : getVolumes()) {
+            log.info("Creating volume '%s'", volume.getName());
+            volService.createVolume(volume);
         }
     }
 

--- a/src/main/java/io/fabric8/maven/docker/VolumeRemoveMojo.java
+++ b/src/main/java/io/fabric8/maven/docker/VolumeRemoveMojo.java
@@ -23,9 +23,13 @@ public class VolumeRemoveMojo extends AbstractDockerMojo {
          throws DockerAccessException, MojoExecutionException  {
       VolumeService volService = serviceHub.getVolumeService();
 
-      for ( VolumeConfiguration volume : getVolumes()) {
-         log.info("Removing volume %s", volume.getName());
-         volService.removeVolume(volume.getName());
+      if(getVolumes() != null){
+         for ( VolumeConfiguration volume : getVolumes()) {
+            log.info("Removing volume %s", volume.getName());
+            volService.removeVolume(volume.getName());
+         }
+      }else{
+         log.info("No volume configuration found.");
       }
    }
 

--- a/src/main/java/io/fabric8/maven/docker/VolumeRemoveMojo.java
+++ b/src/main/java/io/fabric8/maven/docker/VolumeRemoveMojo.java
@@ -21,16 +21,16 @@ public class VolumeRemoveMojo extends AbstractDockerMojo {
    @Override
    protected void executeInternal(ServiceHub serviceHub)
          throws DockerAccessException, MojoExecutionException  {
-      VolumeService volService = serviceHub.getVolumeService();
+       if(getVolumes() == null){
+           log.info("No volume configuration found.");
+           return;
+       }
+       VolumeService volService = serviceHub.getVolumeService();
 
-      if(getVolumes() != null){
-         for ( VolumeConfiguration volume : getVolumes()) {
-            log.info("Removing volume %s", volume.getName());
-            volService.removeVolume(volume.getName());
-         }
-      }else{
-         log.info("No volume configuration found.");
-      }
+       for ( VolumeConfiguration volume : getVolumes()) {
+           log.info("Removing volume %s", volume.getName());
+           volService.removeVolume(volume.getName());
+       }
    }
 
 }

--- a/src/test/java/io/fabric8/maven/docker/VolumeCreateMojoTest.java
+++ b/src/test/java/io/fabric8/maven/docker/VolumeCreateMojoTest.java
@@ -1,0 +1,31 @@
+package io.fabric8.maven.docker;
+
+import io.fabric8.maven.docker.access.DockerAccessException;
+import io.fabric8.maven.docker.service.ServiceHub;
+import io.fabric8.maven.docker.util.Logger;
+import mockit.Injectable;
+import mockit.Mocked;
+import mockit.Tested;
+import mockit.Verifications;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.junit.Test;
+
+public class VolumeCreateMojoTest {
+
+    @Injectable
+    Logger log;
+
+    @Tested(fullyInitialized = false)
+    private VolumeCreateMojo volumeCreateMojo;
+
+    @Mocked
+    ServiceHub serviceHub;
+
+    @Test
+    public void createVolumeGetVolumesReturnsNull() throws DockerAccessException, MojoExecutionException {
+        volumeCreateMojo.executeInternal(serviceHub);
+        new Verifications(){{
+            log.info("No volume configuration found."); times = 1;
+        }};
+    }
+}

--- a/src/test/java/io/fabric8/maven/docker/VolumeRemoveMojoTest.java
+++ b/src/test/java/io/fabric8/maven/docker/VolumeRemoveMojoTest.java
@@ -1,0 +1,31 @@
+package io.fabric8.maven.docker;
+
+import io.fabric8.maven.docker.access.DockerAccessException;
+import io.fabric8.maven.docker.service.ServiceHub;
+import io.fabric8.maven.docker.util.Logger;
+import mockit.Injectable;
+import mockit.Mocked;
+import mockit.Tested;
+import mockit.Verifications;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.junit.Test;
+
+public class VolumeRemoveMojoTest {
+
+    @Injectable
+    Logger log;
+
+    @Tested(fullyInitialized = false)
+    private VolumeRemoveMojo volumeRemoveMojo;
+
+    @Mocked
+    ServiceHub serviceHub;
+
+    @Test
+    public void removeVolumeGetVolumesReturnsNull() throws DockerAccessException, MojoExecutionException {
+        volumeRemoveMojo.executeInternal(serviceHub);
+        new Verifications(){{
+            log.info("No volume configuration found."); times = 1;
+        }};
+    }
+}

